pkg/auth/auth.go

```diff
@@ -99,12 +99,12 @@ func EnsureAuthenticated(client *api.Client, config cfg.Config) bool {
 	}
 
 	var svc Authenticator
-	switch {
-	case config.K8SAuth:
+	switch config.AuthMethod {
```

Comment:
Thinking out loud here: I wonder if we can store a map of registered Authenticator functions (`map[string]Authenticator`), we then could just test the key's presence. And even then in [line 132](https://github.com/cruise-automation/daytona/pull/74/files#diff-c0f4e959d548b59c4ea350952ffcd872183e6c9f5f7dc472bc836e3ec414949bR132) do 

```
type RegisteredAuthenticators map[string]Authenticator

func (ra RegisteredAuthenticators) Available() []string {
    x := make([]string, 0)
    for k, _ := range w {
	x = append(x, k)
    }
    return x
}

log.Fatal().Strs("authMethods", RegisteredAuthenticators.Available()).Msg("You must provide an auth method")
```

This could future proof the validation function.